third_party/move/move-prover/tests/sources/functional/enum.move

```diff
@@ -0,0 +1,86 @@
+module 0x815::m {
+
+    enum CommonFields has copy, drop {
+        Foo{x: u64, y: u8},
+        Bar{x: u64, y: u8, z: u32}
+    }
+
+    spec CommonFields {
+        invariant self.x > 20;
+        invariant (self is CommonFields::Bar) ==> self.z > 10;
```

Comment:
Comments noting which assertions are expected to fail would make this easier to review.

Reply:
done